### ISSUE TITLE
Exception from JAXRS resource is logged without stacktrace

### ISSUE THREAD

**Issue by marm5r** [payara/Payara#2949]
# Description #
----------

Exception stack traces from JAXRS resources are swalloed up by `fish.payara.microprofile.opentracing.jaxrs.JaxrsContainerRequestTracingFilter`
So, there is no chance to trace exceptions back to their origins!

## Expected Outcome

In log file: Exception with stack trace

## Current Outcome

In log file: Exception without stack trace, e.g.:
`[2018-07-08T13:18:46.152+0200] [] [SCHWERWIEGEND] [] [fish.payara.microprofile.opentracing.jaxrs.JaxrsContainerRequestTracingFilter] [tid: _ThreadID=38 _ThreadName=http-thread-pool::http-listener(9)] [timeMillis: 1531048726152] [levelValue: 1000] java.lang.RuntimeException`

## Steps to reproduce (Only for bug reports) 

1. Clone git repository
`git clone https://github.com/marm5r/p5-missing-stacktrace.git`

2. Start application with Payara Micro
```
cd p5-missing-stacktrace
mvn package payara-micro:start
```

3. Make GET request
`curl http://localhost:8080/missing-stacktrace/resources/resource`

4. Examine log output
A RuntimeException from marm5r.p5.sample.Resource is logged without stack trace

## Context ##

Code fragment from JaxrsContainerRequestTracingFilter which swallows up the stack trace:
```
// After method invocation
    @Override
    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) 
            throws IOException {
        // Try block so that we can always attach error info if there is any
        try {
            ...
        } finally {
            // If there's an attached error on the response, log the exception and set the response entity as the 
            // error message (instead of as the error object)
            if (responseContext.hasEntity() && responseContext.getEntity() instanceof Throwable) {
                Throwable throwable = (Throwable) responseContext.getEntity();
                logger.log(Level.SEVERE, throwable.toString());
                responseContext.setEntity(throwable.toString());
            }
        }

    }
```

## Environment ##

- **Payara Version**: 5.182
- **Edition**: Full, Micro
- **JDK Version**: jdk1.8.0_171
- **Operating System**:  Windows

**Comment by MattGill98** [comment]
Hi @marm5r,

Thanks for raising this issue. I've reproduced it, and created internal issue `PAYARA-2921` to address it.

Kind regards,

Matt

**Comment by MattGill98** [comment]
Hi @marm5r,

The bug fix has been resolved for Payara 5.183 and 4.1.2.183. Feel free to try the latest build of master to test the fix. I'll close this issue.

Kind regards,

Matt
